### PR TITLE
Battery widget

### LINKED ISSUE #8: Lock waypoint button when changing control mode to speed and unlock when changing to position

### LINKED ISSUE #9: Change pop-up value when battery is running low

### LINKED ISSUE #13: Change the battery sliders for progressBar

### LINKED ISSUE #15: Change batteries names in the telemetry
Change battery for PC and battery for DVL

### LINKED ISSUE #17: Change battery color based on its level
## Expected Enhancement
Battery color based green/blue , yellow or red depending on battery level

## Current Behavior
Always blue

### LINKED ISSUE #18: Change battery color in function of %
need to change styleSheet

### PR DESCRIPTION
**Pull request by Kam0ss**
## Description
Added a pop-up when one of the battery voltage is running low.
Changed battery appearance for a progressBar instead of a slider and changed its color.
Changed batteries ui sizes.
Blocked the ClearWaypoint button when going in Velocity mode.

## Fixes
Link all the related issues from the issue tracker
- Closes: #15 #13 #9 #8 #18 #17 

## How has this been tested ?
This was tested during test 5 (22/02/2020)

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py [hunk 1]
```diff
@@ -1,9 +1,13 @@
+'''
+\modified by : Camille Sauvain
```
Comment (gauthiermartin) [inline rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py]: This header should be removed

File: rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py [hunk 2]
```diff
@@ -23,44 +27,65 @@ class BatteryWidget(QWidget):
...
-            self.battery_slider.setValue(int(msg.data * 100))
+            self.progressBar.setValue(int(msg.data * 10))
             self.battery_value.setText('{:.2f} V'.format(msg.data))
 
-        if msg.data <= 15.0:
-            nb += 1
-        else:
-            nb = 0
-
+            if msg.data <= float(self.bat_warning):
```
Comment (gauthiermartin) [inline rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py]: Why are we casting to float here ?

File: rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py [hunk 3]
```diff
@@ -23,44 +27,65 @@ class BatteryWidget(QWidget):
...
+                self.progressBar.setStyleSheet('selection-background-color:orange ; background-color:gray')
+            else:
+                self.progressBar.setStyleSheet('selection-background-color:red ; background-color:gray')
+            
         if self.time is not None and rospy.get_time() - self.time >= 60:
             self.nb_msg = 0
 
-        if msg.data <= 15.0 and self.nb_msg == 0 and nb == 20:
-            nb = 0
+        if msg.data <= float(self.bat_warning) and self.nb_msg == 0 and self.nb == 5:
```
Comment (gauthiermartin) [inline rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py]: Why are we casting to float here

File: rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py [hunk 4]
```diff
@@ -23,44 +27,65 @@ class BatteryWidget(QWidget):
...
+            self.nb = 0
             self.time = rospy.get_time()
             self.nb_msg = 1
             root = Tk()
```
Comment (gauthiermartin) [inline rqt_toolbar/src/rqt_toolbar/ToolbarBatteryWidget.py]: We should consider using QTMessageBox https://www.tutorialspoint.com/pyqt/pyqt_qmessagebox.htm

**Review by gauthiermartin — CHANGES_REQUESTED** [review]
Some changes are required

**Review by Kam0ss — COMMENTED** [review]
Changes asked are done.

**Review by gauthiermartin — APPROVED** [review]
LGTM 👍

**Pull request merged**